examples/v1/firmware/firmware.go

```diff
@@ -0,0 +1,62 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"os"
+	"time"
+
+	"github.com/bmc-toolbox/bmclib"
+	"github.com/bmc-toolbox/bmclib/devices"
+	"github.com/bombsimon/logrusr/v2"
+	"github.com/sirupsen/logrus"
+)
+
+func main() {
+	ctx, cancel := context.WithTimeout(context.Background(), 120*time.Second)
+	defer cancel()
+
+	// set BMC parameters here
+	host := ""
+	port := ""
+	user := ""
+	pass := ""
+
+	l := logrus.New()
+	l.Level = logrus.TraceLevel
+	logger := logrusr.New(l)
+
+	if host == "" || user == "" || pass == "" {
+		log.Fatal("required host/user/pass parameters not defined")
+	}
+
+	cl := bmclib.NewClient(host, port, user, pass, bmclib.WithLogger(logger))
+
+	err := cl.Open(ctx)
+	if err != nil {
+		log.Fatal(err, "bmc login failed")
+	}
+
+	defer cl.Close(ctx)
+
+	for _, update := range []string{"/tmp/E6D4INL2.09C.ima"} {
```

Comment:
want to make this a flag value?

Reply:
[edb3e1c](https://github.com/bmc-toolbox/bmclib/pull/261/commits/edb3e1c91cc22a5c38f74e59cce18139409374a0)